enhancements/user-workload-monitoring.md

```diff
@@ -0,0 +1,275 @@
+as well as the newly added Prometheus servers.
+
+The cluster Prometheus Operator (PO) continues to operate the cluster-level Prometheus.
+We will continue to use a centralized Alertmanager (A) cluster.
+The existing Alertmanager cluster will aggregate both user workload alerts as well as cluster alerts. Tenancy will be achieved by forcing specific labels onto metrics and alerts by which the central Alertmanager config can route similarly to how tenancy is achieved at the query level (see below).
+
+Querying data from a single, multi-tenant interface is done using the Thanos Querier (TQ) component.
+Tenancy is enforced at the prometheus query layer. This is achived by leveraging the existing topology using kube-rbac-proxy [1] and prom-label-proxy [2].
+
+Differences between cluster- and user-level is mainly upgrade guarantees. The cluster-level Prometheus stack is an integral part of OpenShift and its reliability is important, keeping the stack immutable and fully testable by us allows us to rely on it to a higher degree. As we have no such possibility for user defined configurations and scrape targets, we can't provide the same guarantees.
+
+There are edge cases where an automated upgrade is not possible in the face of user workload monitoring.
+If users declare i.e. queries in ServiceMonitors which break in newer Prometheus versions
+we most likely cannot perform an automated migration. Such migration without knowledge of user intention behind each query would be exceptionally difficult.
+
+Instead, the "upgradable" condition in the ClusterOperator will be set to false.
+Routing configurations for alerts to be sent to specific teams need to be communicated/configured by an infrastructure owner using the existing Alertmanager stack that is part of cluster-monitoring.
+We will look into extension mechanisms post GA.
+With any OpenShift deployment, we only install the cluster-level part
+and allow infrastructure owner to "enable" the addition of user workload monitoring.
+
+### User Stories
+
+
+#### US1
+
+As a service owner, I’d like to configure a ServiceMonitor CRD so that Red Hat’s supported monitoring solution is able to scrape metrics for my service.
+
+#### US2
+
+As a service owner, I’d like to configure alerting rules for my service so that Red Hat’s supported monitoring solution notifies me if problems turn up.
+
+#### US3
+
+As a member of the operations team, I want to configure Prometheus one time so that all existing and any new namespaces will have access to user workload monitoring.
+
+#### US4
+
+As a member of the operations team, I'd like to add my own exporters so that I can monitor infrastructure services that are not monitored out-of-the-box.
+
+Note: This covers adding ServiceMonitor and PrometheusRule CRDs.
+
+#### US5
+
+As a developer working on a Red Hat product not part of the infrastructure, I’d like to integrate my metrics into a single, supported Prometheus stack so that they are available for my customers.
```

Comment:
How is different from US1, US2 and US4? Is there a difference between "member of the operations team", "service owner" and "developer working on a Red Hat product not part of the infrastructure" when it comes to defining Prometheus targets and rules?

Reply:
cc'ing @sichvoge for clarification.